### PR TITLE
Add name and description for maintenance window targets

### PR DESCRIPTION
**Pull request by nbrys**
Add name and description for maintenance window targets

This pull requests addresses issue 6471

### CODE REVIEW

File: aws/validators.go [hunk 1]
```diff
@@ -1580,6 +1580,19 @@ func validateAwsSSMMaintenanceWindowTaskName(v interface{}, k string) (ws []stri
 	return
 }
 
+func validateAwsSSMMaintenanceWindowTargetName(v interface{}, k string) (ws []string, errors []error) {
+	// https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_RegisterTaskWithMaintenanceWindow.html#systemsmanager-RegisterTaskWithMaintenanceWindow-request-Name
+	value := v.(string)
+
+	if !regexp.MustCompile(`^[a-zA-Z0-9_\-.]{3,128}$`).MatchString(value) {
+		errors = append(errors, fmt.Errorf(
+			"Only alphanumeric characters, hyphens, dots & underscores allowed in %q: %q (Must satisfy regular expression pattern: ^[a-zA-Z0-9_\\-.]{3,128}$)",
```
Comment (nywilken) [inline aws/validators.go]: I think we can shorten this message to 

```
errors = append(errors, fmt.Errorf("%q is not a valid value for %q (only alphanumeric characters, dots, underscores and hyphens are allowed)", value, k))
```

Comment (nywilken) [reply aws/validators.go]: Actually looking at this function a little further it looks like you can just use the function `validateAwsSSMName` defined on line 1157 and update the error message there.

**Comment by ryndaniels** [comment]
@nbrys since it's been a while since we've seen any update on this, we're going to pick up where you left off to get this finished up. Thanks so much for this contribution!

**Comment by ryndaniels** [comment]
To avoid force-pushing to @nbrys' master branch, opened up #8671 with this commit and my updates instead, with both of us getting contribution credit.

**Pull request closed without merging**